the-collab-lab/tcl-17-smart-shopping-list
# Bl add color theme

## Description
This PR will add a Material UI Theme, implement the useStyles and makeStyles functions and make it possible to set colors within components using simple references such as `red`, `blue` and `salmon`, etc. eliminating the need to use hex codes and making it easy to implement consistent styling accross the entire application.



## Related Issue
[#40 ](https://github.com/the-collab-lab/tcl-17-smart-shopping-list/issues/40)


## Acceptance Criteria
- [ ] Colors and contrast must meet accessibility guidelines / pass accessibility tests
[WCAG Color Contrast Testing Site](https://accessible-colors.com/)
- [ ] Material UI Theme is implemented so that colors are easily accessed and remain consistent throughout the application.



## Type of Changes



|     | Type                       |
| --- | -------------------------- |
|  ✓ | :sparkles: New feature     |


## Updates
### Before
<img width="554" alt="before-theme" src="https://user-images.githubusercontent.com/36529777/109399630-d672c380-78f8-11eb-8edb-1a662e82fa0b.PNG">


### After
<img width="411" alt="after-theme" src="https://user-images.githubusercontent.com/36529777/109399900-7bda6700-78fa-11eb-9f17-8f160583e188.PNG">



## How to use the theme colors in your components
Access the theme colors by setting the `className` attribute on an element and setting its value to `{classes.colorYouWantToAccess}`, see example below : 
<img width="270" alt="using-mui-theme" src="https://user-images.githubusercontent.com/36529777/109399866-39188f00-78fa-11eb-8ce5-637a6c667cca.PNG">

## Testing Steps / QA Criteria
*The Material UI Theme is not currently applied to any specific element, so if you would like to test this feature:*
* From your terminal, pull down this branch with `git pull origin BL-add-color-theme` and check that branch out with `git checkout BL-add-color-theme`
* `npm start` to launch the app
* Start a list and navigate to the List view
* Copy the following: `<h1 className={classes.salmon}>IS THIS WORKING</h1>`
* Paste the line of code above inbetween the `<Header />` and the `{loading ? ` , depicted in the image below as line 163
* <img width="290" alt="list-view" src="https://user-images.githubusercontent.com/36529777/109399870-3fa70680-78fa-11eb-99fa-95ec05faae34.PNG">

* Refresh the list page to confirm that `THIS IS WORKING` appears and the color is salmon '#f98365'
